xapian-core/tests/harness/backendmanager.cc

```diff
@@ -123,16 +125,32 @@ BackendManager::get_database(const std::string &dbname,
     string dbleaf = "db__";
     dbleaf += dbname;
     const string& path = get_generated_database_path(dbleaf);
+
+    // checking if a remote backendmanager is being used
+    bool is_remote_bm = false;
+    if (startswith(get_dbtype(), "remote")) {
+	is_remote_bm = true;
+    }
+
     if (path.empty()) {
 	// InMemory doesn't have a path but we want to support generated
 	// databases for it.
+	if (is_remote_bm) {
+	    Xapian::WritableDatabase wdb =
+		get_writable_database_from_sub_manager(dbleaf);
+	    gen(wdb, arg);
+	    return get_remote_database_by_name(dbleaf, 30000);
+	}
```

Comment:
Here we're only dealing with the inmemory backend (that's the only one where `path.empty()` will be true, so I don't think this code would ever get used...

Reply:
I think you want to this lower down so we don't pass a remote WritableDatabase to the generator function (which will work, but be unnecessarily slower).

Reply:
Oh I misread the code, I apologize for the mistake.
get_writable_database_from_sub_manager's purpose was exactly that what you have written.